File: kiyoshi.cs
using System;
using System.Collections.Generic;
using System.Threading;

class Program{
	static readonly Random rand=new Random();

	static void Main(){
		string[] ptn={"ズン","ドコ"};
		const string fin="キ・ヨ・シ!";
		var log=new LinkedList<string>(new[]{"","","","",""});
		const string ans="ズンズンズンズンドコ";

		for(;;){
			var last=ptn[rand.Next(ptn.Length)];
			Console.Write(last);
			log.RemoveFirst();
			log.AddLast(last);
			if(ans==string.Join("",log)) break;
			Thread.Sleep(200);
		}
		Console.WriteLine(fin);
	}
}

File: kiyoshi.vb
Option Strict On
Option Infer On
Imports System.Collections.Generic
Imports System.Threading

Module Program
	ReadOnly rand As New Random()

	Sub Main()
		Dim ptn={"ズン","ドコ"}
		Const fin="キ・ヨ・シ!"
		Dim log As New LinkedList(Of String)({"","","","",""})
		Const ans="ズンズンズンズンドコ"

		Do
			Dim last=ptn(rand.Next(ptn.Length))
			Console.Write(last)
			log.RemoveFirst()
			log.AddLast(last)
			If ans=string.Join("",log) Then Exit Do
			Thread.Sleep(200)
		Loop
		Console.WriteLine(fin)
	End Sub
End Module
